examples/distributed_tree_driver/distributed_tree_driver.cpp

```diff
@@ -350,10 +377,34 @@ int main_(std::vector<std::string> const &args)
 int main(int argc, char *argv[])
 {
   MPI_Init(&argc, &argv);
-  Kokkos::initialize(argc, argv);
 
-  std::cout << "ArborX version: " << ArborX::version() << std::endl;
-  std::cout << "ArborX hash   : " << ArborX::gitCommitHash() << std::endl;
+  const MPI_Comm comm = MPI_COMM_WORLD;
+  int comm_rank;
+  MPI_Comm_rank(comm, &comm_rank);
+  if (comm_rank == 0)
+  {
+    std::cout << "ArborX version: " << ArborX::version() << std::endl;
+    std::cout << "ArborX hash   : " << ArborX::gitCommitHash() << std::endl;
+  }
+
+  // Strip "--help" and "--kokkos-help" from the flags passed to Kokkos if we
+  // are not on MPI rank 0 to prevent Kokkos from printing the help message
+  // multiply.
+  auto kokkos_argv = argv;
+  auto kokkos_argc = argc;
+  if (comm_rank != 0)
+  {
+    auto help_it = std::find_if(kokkos_argv, kokkos_argv + kokkos_argc,
+                                [](std::string const &x) {
+                                  return x == "--help" || x == "--kokkos-help";
+                                });
+    if (help_it != kokkos_argv + kokkos_argc)
+    {
+      std::rotate(kokkos_argv, help_it + 1, kokkos_argv + kokkos_argc);
+      --kokkos_argc;
+    }
```

Comment:
Why did you decide to rotate rather than swapping with the last argument?

Reply:
Swapping now.

Reply:
I just wasted a bunch of time checking that `swap(x, x)` was safe (will happen if you only have the help flag passed has argument) :/

Anyway that's fine